### DISCUSSION TITLE
Merging drives/charges

### DISCUSSION THREAD

**Discussion by krezac** [teslamate-org/teslamate#237] [Ideas]
Hi,
this is by no means critical thing ;)

It would be nice to be able to merge "artifficialy splitted" drives ( i.e. when I stop for coffee/pee/pick up someone). 
The same for charges (updating teslamate while car is charging splits the charge into two).
If I expand this idea, another level could be having entity "trip" with multiple drives and charges.
All of above either manually or "automatically" (end date and start date differs by less than 5 minutes or so).
Thanks for considering. Maybe I'll learn elixir one day...

**Comment by adriankumpf** [comment]
I thought about this in the past and leaned more towards introducing a new entity, like you proposed. But I later postponed the idea because I didn't see a clear benefit anymore. 

That's why I'm curios – which information would be useful to you if there was a "trip" entity that combined multiple drives? Stats like average speed/consumption would not be that meaningful, I guess?

Would it make sense to create a new dashboard "trips" that only lists these or would you prefer that trips completely replace their containing drives to get a "clean" drive history?

**Reply by papakwstas** [reply]
I believe that this addition is highly essential and is already found very helpful on Teslafi as I saw. Whenever I stop for some minutes my trip in order to charge and continue , teslamate cuts the whole trip in parts . Also if I put on the handbrake for just few seconds the same thing happens again. So data remains split and very difficult to merge manually.

**Comment by krezac** [comment]
@adriankumpf what bugs me personally is that I like to see the altitude profile and it's splitted into multiple drives/graphs. The idea of "trip" entity came later (mostly for convenience, if you surf thru data from vacation, there would be less records on trip level. 

But I understand the messed up stats could be more pain for majority of users. I'd say ignore it for now, I'll keep it for long winter nights to suggest/code some acceptable approach.

**Comment by timguy99** [comment]
This would certainly help in cases like mine where I stopped to open the gate to my residence and I have an entry like this. I certainly wouldn't consider this a separate trip. It is most definitely part of my drive home:

`

Start | End | Start | Destination | Duration | km | TR Loss | Ø Speed | Ø Consumption | Consumed | Efficiency | Ø Outdoor T. | Ø Indoor T. | max Speed | max Power | min Power | Ø Power | TR Start | TR End | km Start | km End
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
2019-11-27 16:39:09 | 2019-11-27 16:41:02 | Home | Home | 2 min | 0.15 km | 0 km | 4 km/h | -


`

**Comment by cwanja** [comment]
Not sure how complicated this is, but could you present a dashboard in TeslaMate that offers the ability to merge trips? Simple table view, like Grafana, but with less rows displayed. Maybe just the start, end and distance. Checkbox individual trips and merge. 

Drop the start, end time, mileage in the middle, etc. and have the computed values still completed in Grafana. Just thinking out loud.

**Comment by qwijib0** [comment]
My use case for this would be a little bit different-- but I think it falls in the same general category.  I semi-regularly take a 200 mile or so trip during which there is very little cell coverage in the middle, so I end up with a ~60 mile drive, a few 5 milers where the car saw cell for a moment, then a ~10 mile drive at the end.  I wish there was a way to make them a "journey" where the computed consumption/efficiency, temperature, etc, could be estimated from start/end data values.

**Comment by abraha2d** [comment]
Not sure what happened today, but I ended up with portions of my drive back home chopped up into tiny segments.

![image](https://user-images.githubusercontent.com/5976139/137037224-40effdee-ad78-44cf-ade7-45a09d553fb8.png)

I came looking to see if there was a way to merge multiple drives together, and found this discussion.

I agree with timguy99, my drive history is cluttered with lots of Home -> Home entries (for example if I stop at my mailbox to get mail, or if I stop at the front door to let people out, before going to the garage).

Being able to merge multiple drives would help keep the history clean, and also help avoid the artificial inflation of Home in drive statistics.

**Comment by timguy99** [comment]
Seems like this issue has gotten worse in the last little while. Not sure exactly how long but certainly recently.

**Comment by natrlhy** [comment]
I'm seeing some charges get broken up for some reason (no outages that I've seen at home). Is there a way to merge charges?

![image](https://user-images.githubusercontent.com/19696424/153661599-02c24e88-f0b8-402d-a7fc-557a15bd3916.png)

**Reply by cwanja** [reply]
The only way it "writes" it to the database is once it completes or is interrupted - from the charger or the car perspective. I would assume one of those two scenarios is what happened. Also looks like it was long enough that maybe there was a software update in-between the two which would also pause charging.

**Reply by natrlhy** [reply]
Yeah no updates that I've seen car wise. I'll monitor if this keeps happening.

**Comment by BabyYeggie** [comment]
Anyone know the last version to record drives properly? My last complete drive occurred over a month ago using version 1.26.1 but even that version is broken now. I wonder if the NA servers are simply overwhelmed.

**Reply by krezac** [reply]
My yesterday drive recorded properly (v1.26.1)

**Reply by cwanja** [reply]
I also have not had any issues with the drives or charging sessions.

**Comment by bbock** [comment]
I had some weird problem today that generated over 40 drives for a short 5km trip, each drive having only 1 or 2 positions in the database. It was a one-off problem, everything worked fine for other drives afterwards. I suspect some Tesla API issue, as other API clients got strange data for this specific drive as well.

I wanted to have a clean history of drives as I use it as vehicle log for tax purposes, so I created a SQL function to merge the drives in the database. You can find it [as Gist here](https://gist.github.com/bbock/71fe9d4d02044858555823bc0ed3a1f3).

This was only used a few times and not tested thoroughly, so please do yourself a favor and make a database backup first if you want to use the function.
